### PR TITLE
(#13) Support arguments in run.py

### PR DESCRIPTION
**Pull request by xavierhardy**
Fix binding to all interfaces by default (security issue)
Support host, port, debug, and threaded

### CODE REVIEW

File: run.py [hunk 1]
```diff
@@ -1,4 +1,33 @@
...
+    )
+    parser.add_argument(
+        "--host", "-H", type=str, default="127.0.0.1", help="The interface to bind to."
+    )
+    parser.add_argument(
+        "--port", "-p", type=int, default=5000, help="The port to bind to."
+    )
+    parser.add_argument(
+        "--debug",
+        "--debugger",
```
Comment (mingrammer) [inline run.py]: I think `--debugger` would not be needed. `--debug` and `-d` are enough.

File: run.py [hunk 2]
```diff
@@ -1,4 +1,33 @@
...
+        "-d",
+        action="store_true",
+        default=False,
+        help="Enable the debugger.",
+    )
+    parser.add_argument(
+        "--threaded",
+        "--with-threads",
```
Comment (mingrammer) [inline run.py]: Here too, `--threaded` and `-t` are enough I think.

File: run.py [hunk 3]
```diff
@@ -1,4 +1,33 @@
...
+        "-t",
+        action="store_true",
+        default=False,
+        help="Enable multi threading.",
+    )
+    arguments = parser.parse_args()
+    app.run(**arguments.__dict__)
```
Comment (mingrammer) [inline run.py]: How about use `**vars(arguments)` instead of `__dict__`? :)

Comment (xavierhardy) [reply run.py]: Sure

File: run.py (hunk 2 above)
Comment (xavierhardy) [reply run.py]: OK, I'll change it. Just FYI, these are `--with-threads` and `--debugger` are from the `flask run` CLI.

```
[xhardy@IT-IRL00950 pyreportcard (issue-13)]$ flask run --help
Usage: flask run [OPTIONS]

  Run a local development server.

  This server is for development purposes only. It does not provide the
  stability, security, or performance of production WSGI servers.

  The reloader and debugger are enabled by default if FLASK_ENV=development
  or FLASK_DEBUG=1.

Options:
  -h, --host TEXT                 The interface to bind to.
  -p, --port INTEGER              The port to bind to.
  --cert PATH                     Specify a certificate file to use HTTPS.
  --key FILE                      The key file to use when specifying a
                                  certificate.
  --reload / --no-reload          Enable or disable the reloader. By default
                                  the reloader is active if debug is enabled.
  --debugger / --no-debugger      Enable or disable the debugger. By default
                                  the debugger is active if debug is enabled.
  --eager-loading / --lazy-loader
                                  Enable or disable eager loading. By default
                                  eager loading is enabled if the reloader is
                                  disabled.
  --with-threads / --without-threads
                                  Enable or disable multithreading.
  --help                          Show this message and exit.
```

Comment (mingrammer) [reply run.py]: Oh, I did not recognize that `--debugger` and `--with-threads` is from `flask run`. So .. in this case, using them might be better? How do you think about this?

File: run.py (hunk 3 above)
Comment (mingrammer) [reply run.py]: Thank you

File: run.py (hunk 2 above)
Comment (xavierhardy) [reply run.py]: As you wish, I don't have any strong opinion about that.

Comment (mingrammer) [reply run.py]: Then, use `-d`/`--debugger` and `-t`/`--with-threads` for each. Sorry for the confusion.

Comment (xavierhardy) [reply run.py]: What about `--debug` and `--threaded`?

Comment (mingrammer) [reply run.py]: That's ok too. I just want to use only one long and one short option, not more than 2.

Comment (xavierhardy) [reply run.py]: OK, as I've already changed it to `--debug` and `--threaded`, let's not change it again. Thanks for your feedback.

**Comment by mingrammer** [comment]
Thank you

**Pull request merged**